pkg/util/github_test.go

```diff
@@ -233,3 +237,49 @@ func TestParseGithubURL(t *testing.T) {
 		})
 	}
 }
+
+func TestGithubURL_URL(t *testing.T) {
+	tests := []struct {
+		name      string
+		githubURL GithubURL
+		want      string
+	}{
+		{
+			name: "github tree with no subdir",
+			githubURL: GithubURL{
+				Owner: "o",
+				Repo:  "r",
+				Ref:   "master",
+			},
+			want: "github.com/o/r/tree/master/",
+		},
+		{
+			name: "github tree with a subdir",
+			githubURL: GithubURL{
+				Owner:  "o",
+				Repo:   "r",
+				Ref:    "master",
+				Subdir: "something/maybe",
```

Comment:
General comment on test cases (I don't think this needs a fix, just something I prefer):
IMO it doesn't make sense to have an input overlap between all test cases (like Owner/Repo/Ref does here) because then you're not actually testing that input (as hardcoding would work in that case)

Either things are explicitly shared and shouldn't be in the test case definition, or they shouldn't be the same across all cases.